### ISSUE TITLE
Getting "Template directory must be inside of the repository directory." error when not specifying template_dir option.

### ISSUE THREAD

**Issue by Wiley-Black** [python-semantic-release/python-semantic-release#994]
### Description
Running "semantic-release --noop version --print" to try and verify my configuration and see the next version number.

I have no `template_dir` in my pyproject.toml, so using the default template.

### Expected behavior
Would not expect this error message since I'm using the default templates and don't have a template directory specified.

### Actual behavior
The following error:

```
🛡  You are running in no-operation mode, because the '--noop' flag was supplied
[18:51:22] WARNING  [semantic_release.cli.config] WARNING config.from_raw_config: Token value is missing!                                                              config.py:558
Template directory must be inside of the repository directory.
```

### Environment

Windows 11
Python 3.9
Semantic-Release 9.8.6
Poetry 1.8.2

### Configuration

<details>
<summary>Semantic Release Configuration</summary>

```toml
...
[tool.semantic_release]
version_variables = [
  "src/project/__init__.py:__version__"
]
version_toml = ["pyproject.toml:tool.poetry.version"]
changelog_file = "CHANGELOG.md"
build_command = "poetry build"
dist_path = "dist/"
upload_to_vcs_release = false
upload_to_pypi = false
remove_dist = false
patch_without_tag = true
major_on_zero = true

[tool.semantic_release.commit_parser_options]
allowed_tags = ["build", "chore", "ci", "docs", "feat", "fix", "perf", "style", "refactor", "test"]
minor_tags = ["feat"]
patch_tags = ["fix", "perf"]

[tool.semantic_release.branches.main]
match = "develop"
prerelease = false

[tool.semantic_release.branches.others]
match = "^(?!develop$)"
prerelease = true
prerelease_token = "alpha"
```

</details>
<br>

### Execution Log

<details>
<summary><code>semantic-release -vv <i>command</i></code></summary>

```log
[18:52:55] DEBUG    [semantic_release.cli.commands.main] DEBUG main.main: logging level set to: DEBUG                                                                    main.py:123
🛡  You are running in no-operation mode, because the '--noop' flag was supplied
           DEBUG    [semantic_release.cli.commands.main] DEBUG main.main: global cli options: GlobalCommandLineOptions(noop=True, verbosity=2,                           main.py:135
                    config_file='pyproject.toml', strict=False)
           INFO     [semantic_release.cli.util] INFO util.load_raw_config_file: Loading configuration from pyproject.toml                                                 util.py:79
           DEBUG    [semantic_release.cli.util] DEBUG util.load_raw_config_file: Trying to parse configuration pyproject.toml in TOML format                              util.py:82
           DEBUG    [semantic_release.cli.config] DEBUG config.select_branch_options: Rejecting group 'main' as 'develop' doesn't match 'fix/deploy2'                  config.py:421
           INFO     [semantic_release.cli.config] INFO config.select_branch_options: Using group 'others' options, as '^(?!develop$)' matches 'fix/deploy2'            config.py:414
           DEBUG    [semantic_release.cli.config] DEBUG config.from_raw_config: hvcs token is not set                                                                  config.py:555
           WARNING  [semantic_release.cli.config] WARNING config.from_raw_config: Token value is missing!                                                              config.py:558
Template directory must be inside of the repository directory.
```

</details>

### Additional context

The following [check in cli/config.py:582](https://github.com/python-semantic-release/python-semantic-release/blob/4e59b501374fe6c6f822cc0e9eb41da3e896f642/semantic_release/cli/config.py#L582) triggers the error:

```python
if raw.repo_dir not in template_dir.parents:
```

I added some debug outputs and observe that raw.repo_dir is giving a full path, e.g. "C:\Code\project" whereas the default template_dir is "templates"- not an absolute path.  template_dir.parents then comes out to `['.']`.  I believe the lack of resolving the path to an absolute path might be to blame here.  repo_dir gets resolved to an absolute path by the field_validator in RawConfig, [here](https://github.com/python-semantic-release/python-semantic-release/blob/4e59b501374fe6c6f822cc0e9eb41da3e896f642/semantic_release/cli/config.py#L255).

**Comment by codejedi365** [comment]
I believe this is all because of the path resolution of windows. See #978. I believe this is a more investigated duplicate.

**Comment by Wiley-Black** [comment]
@codejedi365 Yes, just before the "Prevent path traversal attack" check is done on template_dir, the same pattern is applied to changelog on [config.py:574](https://github.com/python-semantic-release/python-semantic-release/blob/4e59b501374fe6c6f822cc0e9eb41da3e896f642/semantic_release/cli/config.py#L574):

```python
        # Prevent path traversal attacks
        if raw.repo_dir not in changelog_file.parents:
```

I'm not sure why my changelog_file and changelog_file.parents resolve to absolute paths (they do, I added a print) and template_dir.parents resolves to ['.'], but a simple fix would be to change both of these checks to something like:

```python
        if raw.repo_dir not in [parent.absolute() for parent in changelog_file.parents]:
```

and

```python
        if raw.repo_dir not in [parent.absolute() for parent in template_dir.parents]:
```

I tested this on my system and it clears up the error.

**Comment by Wiley-Black** [comment]
Alternatively, adding absolute() when first setting up the changelog_file and template_dir resolves it for me as well:

```python
        changelog_file = Path(raw.changelog.changelog_file).expanduser().resolve().absolute()
```

and

```python
        template_dir = Path(raw.changelog.template_dir).expanduser().resolve().absolute()
```

**Comment by Wiley-Black** [comment]
It seems like Windows will find and resolve the absolute path if the file and folder exist.  So my changelog_file is not triggering the error because the file exists.

To anyone looking for a workaround for this issue: add the following section to your pyproject.toml:

```toml
[tool.semantic_release.changelog]
changelog_file = "CHANGELOG.md"
template_dir = "templates"
```

And then make sure that the CHANGELOG.md file exists and the "templates" directory exists in your repo, even if it is empty.

**Comment by codejedi365** [comment]
@Wiley-Black, thank you for all the work you did to isolate this issue for me while I was unavailable and on windows that I don't have ready access to. I will try to get your fix in today

**Comment by codejedi365** [comment]
### :tada: This resolution has been included in version 9.8.8 :tada:

The release is available on:
- [PyPi package](https://pypi.org/project/python-semantic-release/9.8.8/)
- [GitHub release](https://github.com/python-semantic-release/python-semantic-release/releases/tag/v9.8.8)

**Comment by codejedi365** [comment]
@Wiley-Black, thank you for your work in troubleshooting this completely. You made it super easy to make the change. I also paired it with new testing that ensures windows compatibility is checked automatically before release to hopefully not have another issue like this pop up.
